pkg/marker/marker.go

```diff
@@ -110,6 +117,7 @@ func Update(nodeName string) error {
 				Op:   "remove",
 				Path: fmt.Sprintf("/status/capacity/%s~1%s", resourceNamespace, reportedResource),
 			})
+			delete(reportedResourcesCache, reportedResource)
```

Comment:
This is not correct. You are updating the cache before we get to apply the patch and get confirmation about its success.